sdk/resource/host_id.go

```diff
@@ -38,6 +38,7 @@ type fileReader func(string) (string, error)
 
 type commandExecutor func(string, ...string) (string, error)
 
+// nolint: unused  // This is used by the hostReaderBSD, gated by build tags.
```

Comment:
I am not 100% sure about this. I can build and the tests are passing with flying colors after removing `readFile`, but I am hesitant to remove it.

It seems to be similar to `execCommand` below. See [here](https://github.com/open-telemetry/opentelemetry-go/blob/282a47e3d3d2b7b4d1d79701785e8b8a4537cdb0/sdk/resource/host_id_test.go#L88).

Reply:
Is this failing on your local system?

Reply:
Yes it is, without it I get:
```
resource/host_id.go:41:6: func `readFile` is unused (unused)
func readFile(filename string) (string, error) {
     ^
make: *** [golangci-lint/./sdk] Error 1
```

When running `make`.

Reply:
How about moving the functions to `host_id_bsd.go`? Will it solve the issue?

Reply:
https://github.com/open-telemetry/opentelemetry-go/issues/3972